pkg/api/pod/util.go

```diff
@@ -23,6 +23,18 @@ import (
 	"k8s.io/kubernetes/pkg/util/intstr"
 )
 
+const (
+	// The annotation value is a string specifying the hostname to be used for the pod e.g 'my-webserver-1'
+	PodHostnameAnnotation = "pod.beta.kubernetes.io/hostname"
```

Comment:
To extend on this.  The user can set the pod hostname without setting the subdomain, in which case they won't get the full behavior of crafting a service-centriic FQDN.  I assert that is OK.